### PR TITLE
CCP protocol implementation

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

File: scapy/contrib/automotive/ccp.py [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+
+from scapy.packet import Packet, bind_layers, bind_bottom_up
+from scapy.fields import XIntField, FlagsField, ByteEnumField, \
+    ThreeBytesField, XBitField, ShortField, IntField, XShortField, \
+    ByteField, XByteField, StrFixedLenField, LEShortField
+from scapy.layers.can import CAN
+
+
+class CCP(CAN):
+    name = 'CAN Configuration Protocol'
```
Comment (ebroecker) [inline scapy/contrib/automotive/ccp.py]: I think it's "Can Calibration Protocol"

Comment (polybassa) [reply scapy/contrib/automotive/ccp.py]: Thanks. I was wrong

**Comment by gpotter2** [comment]
Hey @polybassa, some of the ISOTP tests are a bit unstable. Do you think you could have a look ? Thanks !
https://travis-ci.org/secdev/scapy/jobs/495430130

We have a `retry_test` function in `scapy.tools.UTscapy` in last resort, if the failure isn’t related to your test (what we use when pinging servers / doing dns), but I don’t think you should be using it for local tests

**Comment by polybassa** [comment]
Hi @gpotter2 I’ll have a look. Unfortunately the link from you doesn’t give me any information. Could you please specify, which tests are failing the most. 

We also discovered, that high load on Travis makes some of our tests more likely to fail. I wonder if this could be simulated, to find failing tests better.

**Comment by polybassa** [comment]
Just saw your edit. Alright.

**Comment by gpotter2** [comment]
Here’s a failure: https://travis-ci.org/secdev/scapy/jobs/495745915

And here’s a backup of the logs: https://pastebin.com/Fz1fxT0T

**Comment by polybassa** [comment]
Hi @gpotter2. I've created PR #1863. Hope this makes the test more stable.

**Comment by polybassa** [comment]
This PR is complete, from my side.

File: scapy/contrib/automotive/ccp.py [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+        if not hasattr(other, "cmd"):
+            return 0
+
+        new_pl_cls = self.get_dto_cls(other.cmd)
+        if self.payload_cls != new_pl_cls and \
+                self.payload_cls == DEFAULT_DTO:
+            data = bytes(self.load)
+            self.remove_payload()
+            self.add_payload(new_pl_cls(data))
+            self.payload_cls = new_pl_cls
```
Comment (gpotter2) [inline scapy/contrib/automotive/ccp.py]: Could you document a bit what this is doing ?

I don’t think one would expect `answers()` to edit the packets, it is commonly only a test

Comment (polybassa) [reply scapy/contrib/automotive/ccp.py]: Done

**Comment by polybassa** [comment]
I saw this bug with python-can. Is there something I can do for you? Would it be a solution, to use a fixed version of python-can?

**Comment by gpotter2** [comment]
`Python-can` just released a new version. The issue is now fixed :-)

This PR fails the spelling check:
```
scapy/contrib/automotive/ccp.py:562: dependant  ==> dependent
```

**Comment by gpotter2** [comment]
Thanks for your work !

**Comment by polybassa** [comment]
Thanks for your work, as well!

**Pull request merged**